### PR TITLE
DetailsViewController transferred to CollectionView

### PR DESCRIPTION
**Pull request by urutet**

### CODE REVIEW

File: MovieSaver/CustomUIElements/ImageCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    let imageView = UIImageView()
+    
+    imageView.layer.masksToBounds = true
+
+    return imageView
+  }()
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    addSubview(imageView)
```
Comment (wadimbylvar) [inline MovieSaver/CustomUIElements/ImageCollectionViewCell.swift]: `contentView.addSubview`

File: MovieSaver/CustomUIElements/LabelCollectionViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    let label = UILabel()
+    
+    label.font = FontsManager.bold(ofSize: 14)
+    
+    return label
+  }()
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    addSubview(label)
```
Comment (wadimbylvar) [inline MovieSaver/CustomUIElements/LabelCollectionViewCell.swift]: То же самое

File: MovieSaver/CustomUIElements/TextViewCollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    textView.font = FontsManager.regular(ofSize: 14)
+    textView.isSelectable = false
+    textView.isEditable = false
+    
+    return textView
+  }()
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    addSubview(textView)
```
Comment (wadimbylvar) [inline MovieSaver/CustomUIElements/TextViewCollectionViewCell.swift]: То же самое

File: MovieSaver/CustomUIElements/WebViewCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import UIKit
+import WebKit
+
+class WebViewCollectionViewCell: UICollectionViewCell {
+  private let webView = WKWebView()
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    addSubview(webView)
```
Comment (wadimbylvar) [inline MovieSaver/CustomUIElements/WebViewCollectionViewCell.swift]: То же самое

**Review by wadimbylvar — CHANGES_REQUESTED** [review]

File: MovieSaver/CustomUIElements/ImageCollectionViewCell.swift (hunk 1 above)
Comment (urutet) [reply MovieSaver/CustomUIElements/ImageCollectionViewCell.swift]: Done

File: MovieSaver/CustomUIElements/LabelCollectionViewCell.swift (hunk 2 above)
Comment (urutet) [reply MovieSaver/CustomUIElements/LabelCollectionViewCell.swift]: Done

File: MovieSaver/CustomUIElements/TextViewCollectionViewCell.swift (hunk 3 above)
Comment (urutet) [reply MovieSaver/CustomUIElements/TextViewCollectionViewCell.swift]: Done

File: MovieSaver/CustomUIElements/WebViewCollectionViewCell.swift (hunk 4 above)
Comment (urutet) [reply MovieSaver/CustomUIElements/WebViewCollectionViewCell.swift]: Done

File: MovieSaver/Controller/MovieDetailViewController.swift [hunk 5]
```diff
@@ -1,170 +1,187 @@
...
+    case 2:
+      return Constants.numberOfItemsInThirdSection
+    default:
+      return 0
+    }
   }
   
-  public func setMovieWebView(url: URL) {
-    videoWebView.loadRequest(URLRequest(url: url))
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
```
Comment (wadimbylvar) [inline MovieSaver/Controller/MovieDetailViewController.swift:196]: This method is too long

File: MovieSaver/Controller/MovieDetailViewController.swift [hunk 6]
```diff
@@ -1,170 +1,199 @@
...
-  private let yearLabel: UILabel = {
-    let label = UILabel()
-    
-    label.font = Constants.movieYearLabelFont
-    label.textColor = Constants.movieYearTextColor
-    
-    return label
-  }()
+  private let collectionView = UICollectionView(frame: .zero, collectionViewLayout: createLayout())
+  private var movie = Movie(
```
Comment (wadimbylvar) [inline MovieSaver/Controller/MovieDetailViewController.swift:32]: I mentioned this moment on our call

**Review by wadimbylvar — CHANGES_REQUESTED** [review]

File: MovieSaver/Controller/MovieDetailViewController.swift (hunk 6 above)
Comment (urutet) [reply MovieSaver/Controller/MovieDetailViewController.swift:32]: The problem was resolved in branch feature/addMovieViewController-to-mvvm

File: MovieSaver/Controller/MovieDetailViewController.swift (hunk 5 above)
Comment (urutet) [reply MovieSaver/Controller/MovieDetailViewController.swift:196]: Divided into parts in the latest commit

**Review by wadimbylvar — APPROVED** [review]

**Pull request merged**